internals/search.py

```diff
@@ -311,6 +312,73 @@ def _sort_by_total_order(
   return sorted_id_list
 
 
+def make_cache_key(
+  user_query: str, sort_spec: str | None, show_unlisted: bool,
+  show_deleted: bool, show_enterprise: bool, start: int, num: int,
+  name_only: bool) -> str:
+  """Return a redis key string to store cached search results."""
+  return '|'.join([
```

Comment:
I wonder if we can make the key name more readable, since the max allowed key size is quite big [1]? e.g. FeatureSearch||unlisted=None|deleted=True|enterprise=False|...'.

It would also prevent issues if we modify the parameters of a cache key, e.g. changing the sequence of these params, etc.

[1]https://redis.io/docs/latest/develop/use/keyspace/

Reply:
Done